frontend/src/libs/components/icon/icon.tsx

```diff
@@ -0,0 +1,20 @@
+import { type IconName } from '#libs/enums/enums.js';
+import { getValidClassNames } from '#libs/helpers/get-valid-class-names.js';
+import { iconNameToPlainSvgMap } from '#libs/maps/icon-name-to-plain-svg.js';
+
+import styles from './styles.module.scss';
+
+type Properties = {
+  name: IconName;
+  className?: string;
+};
+
+const Icon: React.FC<Properties> = ({ name, className = '' }) => {
+  const SelectedIcon = iconNameToPlainSvgMap[name];
+
+  const cssClass = getValidClassNames(styles[className]);
```

Comment:
From one side you pass `className` prop in order to set styles outside. From other side this class name is used in `styles` object, so you limit yourself only via styles in range of imported scss file.

I think it's 2 ways to make it more consistent: or you receive `className: classValue` (from lib) and pass the whole `style['smth']` outside, or just make prop `className: 'profile-button-background' | 'further-option'` and use these kind of presets. The second seems better to me

Reply:
Let's not pass classes or styles to Icon component. For background color, margin, padding we can wrap Icon component on div where we use it. If we need to change the color of icon, we can pass `color` property

Reply:
@vladyslavdobychin please follow Farid's comment, it's mandatory one